File: typescript/leetcode/0009-palindrome-number/solution01.ts
/**
 * @param {number} x
 * @return {boolean}
 */
export function isPalindrome(x: number): boolean {
  if (x < 0) {
    return false
  }
  const original = x

  let reversed = 0
  while (x > 0) {
    reversed = reversed * 10 + (x % 10)
    x = Math.floor(x / 10)
  }

  return original === reversed
}

File: python/leetcode/0009-palindrome-number/solution01.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        original = x
        reversed = 0
        while x > 0:
            reversed = reversed * 10 + x % 10
            x //= 10

        return original == reversed
